packages/editor-preview/src/browser/editor-preview-widget.ts

```diff
@@ -0,0 +1,226 @@
+        super.onActivateRequest(msg);
+        if (this.editorWidget_) {
+            this.editorWidget_.activate();
+        }
+    }
+
+    protected attachPreviewWidget(w: Widget) {
+        (this.layout as PanelLayout).addWidget(w);
+        this.title.label = w.title.label;
+        this.title.iconClass = w.title.iconClass;
+        this.title.caption = w.title.caption;
+
+        if (Saveable.isSource(w)) {
+            Saveable.apply(this);
+            const dirtyListener = w.saveable.onDirtyChanged(() => {
+                this.pinEditorWidget();
+            });
+            this.toDispose.push(dirtyListener);
+        }
+        w.parent = this;
+        w.activate();
+        this.onDidChangeTrackableWidgetsEmitter.fire([w]);
+    }
+
+    onAfterAttach(msg: Message) {
+        super.onAfterAttach(msg);
+        if (this.editorWidget_ && !this.editorWidget_.isAttached) {
+            this.attachPreviewWidget(this.editorWidget_);
+        }
+        this.addTabPinningLogic();
+    }
+
+    private lastParent: DockPanel;
+
+    protected addTabPinningLogic() {
+        const parent = this.parent;
+        if (!this.pinned_ && parent instanceof DockPanel) {
+            if (!this.lastParent) {
+                this.lastParent = parent;
+            }
+
+            const tabBar = find(parent.tabBars(), bar => bar.titles.indexOf(this.title) !== -1);
+
+            // Widget has been dragged into a different panel
+            if (this.lastParent !== parent || !tabBar) {
+                this.pinEditorWidget();
+                return;
+            }
+
+            const layoutListener = (panel: DockPanel) => {
+                if (tabBar !== find(panel.tabBars(), bar => bar.titles.indexOf(this.title) !== -1)) {
+                    this.pinEditorWidget();
+                }
+            };
+            parent.layoutModified.connect(layoutListener);
+            this.pinListeners.push({dispose: () => parent.layoutModified.disconnect(layoutListener)});
+
+            const tabMovedListener = (w: Widget, args: {title: Title<Widget>}) => {
+                if (args.title === this.title) {
+                    this.pinEditorWidget();
+                }
+            };
+            tabBar.tabMoved.connect(tabMovedListener);
+            this.pinListeners.push({dispose: () => tabBar.tabMoved.disconnect(tabMovedListener)});
+
+            requestAnimationFrame(() => {
+                const tabNode = tabBar.contentNode.children.item(tabBar.currentIndex);
+                const dblClickListener = (event: Event) => this.pinEditorWidget();
+                tabNode.addEventListener('dblclick', dblClickListener);
+                this.pinListeners.push({dispose: () => tabNode.removeEventListener('dblclick', dblClickListener)});
+            });
+        }
+    }
+
+    protected onResize(msg: Widget.ResizeMessage): void {
+        if (this.editorWidget_) {
+            if (msg.width < 0 || msg.height < 0) {
```

Comment:
Have you tried just to forward a message to an editor?
```ts
MessageLoop.sendMessage(this.editorWidget_, msg);
```

Reply:
I updated to pass the message, but we will still need to set the size until autoSizing is working. if you like we can move the code to the editor widget.

Reply:
The way I read / interpreted this is the resize will not be performed here: https://github.com/theia-ide/theia/blob/e8dd128c25294b23741cc345067952ef7cd85864/packages/monaco/src/browser/monaco-editor.ts#L291 due to autoSizing being false.